client/js/serverstate.js

```diff
@@ -100,10 +100,6 @@ export function batchEnd() {
 
 function receiveDelta(delta) {
   // the order of widget changes is not necessarily correct and in order to avoid cyclic children, this first moves affected widgets to the top level
-  for(const widgetID in delta.s)
-    if(delta.s[widgetID] && delta.s[widgetID].parent !== undefined && widgets.has(widgetID))
-      $('#topSurface').appendChild(widgets.get(widgetID).domElement);
-
```

Comment:
Removing this breaks for example FreeCell ( https://test.virtualtabletop.io/PR-1423/freecell ). The 8 columns of widgets are shuffled and each card is a child of the one under it.

So when resetting the game, it basically every time causes a child-parent-relationship to swap. When applying the delta without first moving the `domElement`s to an intermediate parent, it creates circular parent relationships.

Reply:
Ah I see, to restate to make sure I'm looking at the same problem, the root issue is that given:
```html
<div id="a">
  <div id="b"></div>
</div>
```
```js
b.appendChild(a);
```
Will throw `DOMException: Failed to execute 'appendChild' on 'Node': The new child element contains the parent.`
Demo: https://jsbin.com/jejuvud/edit?html,css,js,output

I guess the most straightforward fix would be to give the widget the relative transform in `#topSurface` that it should have to keep it's original from position.

Reply:
Restored this in the latest patch, but adding in preserving the transform relative to the root so that the from position is calculated correctly in `widget.js` when adding to the new parent.